streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2331,484 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+        final String input = "input";
+        final String output = "output";
+        final String storeName = "store";
+
+        builder.stream(input, consumed)
+            .toTable()
+            .mapValues(
+                value -> value.charAt(0) + 48,
+                Materialized.<String, Integer, KeyValueStore<Bytes, byte[]>>as(storeName)
+                    .withKeySerde(Serdes.String())
+                    .withValueSerde(Serdes.Integer()))
+            .toStream()
+            .to(output);
+
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                builder.build(),
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                Instant.ofEpochMilli(0L))) {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, new StringSerializer(), new StringSerializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestOutputTopic<String, Integer> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.Integer().deserializer());
+            final KeyValueStore<String, String> store = driver.getKeyValueStore(storeName);
+            
+            inputTopic.pipeInput("A", "green", 10L);
+            inputTopic.pipeInput("B", "green", 9L);
+            inputTopic.pipeInput("A", "blue", 12L);
+            inputTopic.pipeInput("C", "yellow", 15L);
+            inputTopic.pipeInput("D", "green", 11L);
+
+            final Map<String, Integer> expectedStore = new HashMap<>();
+            expectedStore.putIfAbsent("A", 146);
+            expectedStore.putIfAbsent("B", 151);
+            expectedStore.putIfAbsent("C", 169);
+            expectedStore.putIfAbsent("D", 151);
+
+            assertEquals(expectedStore, asMap(store));
+
+            assertEquals(
+                asList(
+                    new TestRecord<>("A", 151, Instant.ofEpochMilli(10)),
+                    new TestRecord<>("B", 151, Instant.ofEpochMilli(9)),
+                    new TestRecord<>("A", 146, Instant.ofEpochMilli(12)),
+                    new TestRecord<>("C", 169, Instant.ofEpochMilli(15)),
+                    new TestRecord<>("D", 151, Instant.ofEpochMilli(11))),
+                outputTopic.readRecordsToList());
+
+        }
+    }
+
+    private static Map<String, String> asMap(final KeyValueStore<String, String> store) {
+        final HashMap<String, String> result = new HashMap<>();
+        store.all().forEachRemaining(kv -> result.put(kv.key, kv.value));
+        return result;
+    }
+
+    @Test
+    public void shouldPropagateRepartitionFlagAfterGlobalKTableJoin() {
```

Comment:
The test itself is great -- but the name does not really match -- there is no `GlobalKTable` in this test.

Might also be good to pipe some data using the TTD in this test.